test/testdata/parser/quoted_symbol_bug.rb

```diff
@@ -0,0 +1,8 @@
+# typed: true
```

Comment:
minor feedback:

- we don't usually put `_bug` in our test names 
- I would have personally put this in `test/testdata/rewriter/`, because the original crash was caused by an ENFORCE in `rewriter/` and the contents of this test relies on the `prop` DSL.

Up to you whether you'd like to change either of those things.

Reply:
ack, will fix